sensors/omron_d6t/omron_d6t.py

```diff
@@ -0,0 +1,96 @@
+import time, crcmod.predefined
+from sensors.sensor import Sensor
+
+class OmronD6T(Sensor):
+    __START_COMMAND = 0x4c
+    __ARRAY_SIZE = 16
+    __BUFFER_LENGTH = 35
+    __PEC_BUFFER = [0x15]
+    __MAX_CELLS_ABOVE_ROOM_TEMPERATURE = 10
+
+    def __init__(self, bus = 1, address = 0x0a):
+        super().__init__(bus, address, self.__START_COMMAND)
+
+    @property
+    def buffer_length(self):
+        return self.__BUFFER_LENGTH
+
+    @property
+    def occupied_status(self):
+        room_temperature, temperature_matrix = self.__read()
+
+        return self.__temperature_cells_above_threshold(room_temperature, temperature_matrix) >= self.__MAX_CELLS_ABOVE_ROOM_TEMPERATURE
+
+    def __temperature_cells_above_threshold(self, room_temperature, temperature_array):
+        """
+        :param room_temperature: Room temperature
+        :param temperature_array: Temperature list from sensor
+
+        :return: Number of cells above room temperature 
+        """
+        return sum(temperature > room_temperature for temperature in temperature_array)
+
+    def __read(self):
+        """
+        Read sensor data and parse it appropriately. Checks PEC to ensure the data is valid.
+
+        :return: Tuple containing room temperature and an array of measured temperatures, or None if there was an issue
+        reading from the sensor
+        """
+
+        # This loop will run until a valid value is received from the sensor (or max retries is reached)
+        for i in range(0, self.MAX_RETRIES):
+            bytes_read, sensor_raw_data = self.read_raw_data()
+
+            if bytes_read == self.buffer_length:
+                # Read PEC byte for checksum
+                crc_expected = sensor_raw_data[34]
+
+                # Calculate CRC-8 of received bytes (excluding PEC byte at end)
+                crc_actual = self.__check_pec(sensor_raw_data[:34])
+
+                if crc_expected == crc_actual:
+                    # Number of bytes read, and CRC is correct -> Parse Temperature data
+                    return self.__parse_temperature_data(sensor_raw_data)
+                else:
+                    # Wait between retries
+                    time.sleep(0.05)
+                    continue
+            else: # Handle i2c error transmissions
+                print('\n[Error]: Number of bytes mismatch. Number of bytes read: ' + str(bytes_read))
```

Comment:
The [logging library](https://docs.python.org/3/library/logging.html#module-logging) provides out of the box behavior like this.

At the top you would put:
```
import logging
logging.basicConfig(level=logging.INFO)
logger = logging.getLogger('OmronD6T')
```
The basic config level determines what will actually be printed so you can switch between levels or severity.
And then:
```
logger.error(...)
```
and it would output something like:
```
ERROR:OmronD6T:...
```

Reply:
❤️